cmd/dev.go

```diff
@@ -0,0 +1,348 @@
+	promptInput := text.Input{
+		Prompt:      "What is your Project ID?",
+		Placeholder: "",
+		Validator:   projectIDValidator,
+	}
+
+	return text.Run(&promptInput)
+}
+
+func CmdExec(command string, dir string, env []string, name string) {
+	baseCommand, args, _ := strings.Cut(command, " ")
+
+	cmd := exec.Command(baseCommand, strings.Split(args, " ")...)
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	cmd.Dir = filepath.Join(cwd, dir)
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, env...)
+
+	// redirect stdout and stderr to custom output
+	cmd.Stdout = &customOutput{scope: name}
+	cmd.Stderr = &customOutput{scope: name}
+
+	cmd.Run()
+}
+
+type customOutput struct {
+	scope string
+}
+
+// parse the logs and prefix them with the scope
+func (c customOutput) Write(p []byte) (int, error) {
+	normalized := strings.ReplaceAll(string(p), "\r\n", "\n")
+	lines := strings.Split(normalized, "\n")
+
+	for _, line := range lines {
+		fmt.Printf("[%s] %s\n", c.scope, line)
+	}
+
+	return len(p), nil
+}
+
+func formatEnvVars(key string, port string, name string, microType string, hostname string) []string {
+	hostnameEnv := fmt.Sprintf("DETA_SPACE_APP_HOSTNAME=%s", hostname)
+	microNameEnv := fmt.Sprintf("DETA_SPACE_APP_MICRO_NAME=%s", name)
+	microTypeEnv := fmt.Sprintf("DETA_SPACE_APP_MICRO_TYPE=%s", microType)
+
+	keyEnv := fmt.Sprintf("DETA_PROJECT_KEY=%s", key)
+	portEnv := fmt.Sprintf("PORT=%s", port)
+
+	envStrings := []string{hostnameEnv, microNameEnv, microTypeEnv, keyEnv, portEnv}
+
+	return envStrings
+}
+
+// Normalize the path to a subtree path
+func normalizePath(pathString string) string {
+	p := path.Join("/", pathString, "/")
+	if p == "/" {
+		return p
+	}
+
+	return p + "/"
+}
+
+func generatePath(pathString *string, name string, primary bool) string {
+	if pathString == nil {
+		if primary {
+			return "/"
+		}
+
+		return normalizePath(name)
+	}
+
+	return normalizePath(*pathString)
+}
+
+func getDevCommand(micro *shared.Micro) (string, error) {
+	if micro.Dev != "" {
+		return micro.Dev, nil
+	}
+
+	engine, ok := shared.EngineAliases[micro.Engine]
+	if !ok {
+		return "", fmt.Errorf("unsupported engine for %s", micro.Name)
+	}
+
+	command, ok := enginesToDevCommand[engine]
+	if !ok {
+		return "", fmt.Errorf("no dev command specified for %s", micro.Name)
+	}
+
+	return command, nil
+}
+
+func cleanup() {
+	logger.Println("\nCleaning up...")
+}
+
+func dev(cmd *cobra.Command, args []string) error {
+	logger.Println()
+
+	e := make(chan os.Signal)
+	signal.Notify(e, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-e
+		cleanup()
+		os.Exit(1)
+	}()
+
+	// check space version
+	c := make(chan *checkVersionMsg, 1)
+	defer close(c)
+	go checkVersion(c)
+
+	var err error
+
+	// grab to port for the proxy server from the env or default to 8080
+	serverPort := os.Getenv("PORT")
+	if serverPort == "" {
+		serverPort = "8080"
+	}
+	hostname := fmt.Sprintf("http://localhost:%s", serverPort)
```

Comment:
the `DETA_SPACE_APP_HOSTNAME` env variable in micros doesn't inclue the protocol prefix
```suggestion
	hostname := fmt.Sprintf("localhost:%s", serverPort)
```